plugins/wacom-usb/fu-wac-device.c

```diff
@@ -726,6 +725,58 @@ fu_wac_device_add_modules(FuWacDevice *self, GError **error)
 		return FALSE;
 	}
 
+	/* validate versions of each submodule */
+	for (guint8 i = 0; i < buf[3]; i++) {
+		guint8 fw_type = buf[(i * 4) + 4] & ~0x80;
+		guint16 ver;
+
+		if (!fu_memread_uint16_safe(buf,
+					    sizeof(buf),
+					    (i * 4) + 5,
+					    &ver,
+					    G_BIG_ENDIAN,
+					    error))
+			return FALSE;
+
+		/* module is in transitional state or requires re-flashing */
+		if (ver == 0) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_FOUND,
+				    "module %u has error state",
+				    fw_type);
+			return FALSE;
+		}
+	}
+
+	memcpy(out, buf, sizeof(buf));
+	return TRUE;
+}
+
+static gboolean
+fu_wac_device_add_modules(FuWacDevice *self, GError **error)
+{
+	g_autofree gchar *version_bootloader = NULL;
+	guint8 buf[32] = {0xff};
```

Comment:
can we make this a GByteArray? I'd feel a lot more warm-and-fuzzy about passing that to the callback.